### DISCUSSION TITLE
Bus factor: 📈

### DISCUSSION THREAD

**Discussion by CAD97** [pest-parser/pest#664] [Announcements]
Your pest team at the end of July 2022:

## Owners

Can do [all of the things](https://docs.github.com/en/organizations/managing-peoples-access-to-your-organization-with-roles/roles-in-an-organization#permissions-for-organization-roles) to manage the organization itself.

- <img src="https://avatars.githubusercontent.com/u/2410580?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @tomtau
- <img src="https://avatars.githubusercontent.com/u/4136413?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @dragostis
- <img src="https://avatars.githubusercontent.com/u/5992217?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @CAD97

## Admins

Can do [all of the things](https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/repository-roles-for-an-organization#permissions-for-each-role) to manage repositories, and has publish permissions to crates-io.

- <img src="https://avatars.githubusercontent.com/u/2410580?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @tomtau
- <img src="https://avatars.githubusercontent.com/u/4136413?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @dragostis
- <img src="https://avatars.githubusercontent.com/u/5992217?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @CAD97

## Maintainers

Can merge PRs.

- <img src="https://avatars.githubusercontent.com/u/603677?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @NoahTheDuke
- <img src="https://avatars.githubusercontent.com/u/2410580?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @tomtau
- <img src="https://avatars.githubusercontent.com/u/5992217?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @CAD97

## Triage

Can manage issues.

- <img src="https://avatars.githubusercontent.com/u/94953?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @glyn
- <img src="https://avatars.githubusercontent.com/u/603677?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @NoahTheDuke
- <img src="https://avatars.githubusercontent.com/u/1239091?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @nathanielknight
- <img src="https://avatars.githubusercontent.com/u/2410580?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @tomtau
- <img src="https://avatars.githubusercontent.com/u/3161395?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @jhpratt
- <img src="https://avatars.githubusercontent.com/u/5992217?v=4" alt="profile picture"  height="25" align="middle" border-radius="50%"/> @CAD97

🥰 Thank you to everyone who stepped up, and here's to pest staying maintained for a while more 🥂

I'll still be around and help out where I can, but now the project is no longer exclusively blocked on me finding time 🎉

**Comment by CAD97** [comment]
One further note on maintainer continuity:

Everyone on this list should feel free to remove themselves if they don't have time/desire to participate anymore. If it's short-term, setting your [status as busy](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/personalizing-your-profile#setting-a-status) will avoid assigning you when a team is assigned. To scope this to a team and not GH-global, ask a team manager to set you up in the "do not assign" list.

The main goal here is to have enough people to enable passive upkeep maintenance/support of pest to have staying power. Please *do* invite more people into teams when they offer to help out; pest is the community's project now.

**Comment by tomtau** [comment]
Hey everyone,  I'm excited to join the effort in continuing pest's maintenance! 🥂

First off, given the last release was over two years ago (https://github.com/pest-parser/pest/issues/491 https://github.com/pest-parser/pest/issues/485 ), I'll try to cut a new release by this weekend. @CAD97 mentioned that all the previous releases were done manually by @dragostis, so I'm not sure of the proper release process here. Anyway, I'll draft something automated, most likely via this GH Action: https://github.com/katyo/publish-crates and we'll see if it works.

I assume the core maintenance effort will revolve around the pest repo and upkeeping the current codebase, but more help is welcome in other repos:

- "[book](https://github.com/pest-parser/book)" + "[site](https://github.com/pest-parser/site)": the book has a few items to fix and sections to complete, and the site's embedded editor may need some work, etc. On the previous thread, @maksugr and @yarcowang mentioned they would like to help, but were new to Rust and pest, so taking care of these two repos could work. 

- "[pest-fmt](https://github.com/pest-parser/pest-fmt)" + "[intellij-pest](https://github.com/pest-parser/intellij-pest)" + "[pest.vim](https://github.com/pest-parser/pest.vim)": these repos were primarily maintained by @dbrgn, @oovm, @ice1000, and @sbeckeriv, but I'm not sure if any of them has the time or desire to maintain these repos anymore.

- "[pest3](https://github.com/pest-parser/pest3)" + "[ast](https://github.com/pest-parser/ast)": this effort is currently stalled, but we could revisit it after the pest repo maintenance is more in order.

In any case, if you're interested in helping with the above repos, please reply to this thread and @ me on any issues or PRs in those repos which need a follow-up. Thank you, everyone, and I'm looking forward to keeping pest alive with y'll! 🎉

**Reply by oovm** [reply]
I no longer develop pest-fmt, fmt may be integrated into intellij-pest

**Reply by ice1000** [reply]
I am still willing to maintain intellij-pest if time allows.

**Reply by NoahTheDuke** [reply]
I can help with or take over `pest.vim` as necessary.

**Reply by CAD97** [reply]
I just added the admin team to publishers of `from-pest` and `pest-ast`.

**Reply by dbrgn** [reply]
I might be able to help with issues in `pest.vim`, but might be quite passive depending on available time (I don't actively work on a project with pest right now). If other people (e.g. @NoahTheDuke) want to help out or take over, that would be welcome.

**Reply by nathanielknight** [reply]
I'd also be happy to pitch in on the site. I've been working on a branch that fixes the "share" button, but I wasn't able to get the existing build working. I'm not sure whether I should hack together a solution by scraping the existing site or try to contribute to a more comprehensive re-work.

**Reply by tomtau** [reply]
Thank you all for the interest and information!

@oovm if that's the case, could you help to add @ice1000 or the @pest-parser/intellij-pest team as owners to the [pest_fmt](https://crates.io/crates/pest_fmt) crate? https://doc.rust-lang.org/cargo/reference/publishing.html#cargo-owner
```
cargo owner --add github:pest-parser/intellij-pest
```

@NoahTheDuke I added you to the `pest.vim` team. @dbrgn @NoahTheDuke do you think you'll want to integrate `pest_fmt` into `pest.vim`?

@nathanielknight currently, the site ([pest-parser.github.io](https://github.com/pest-parser/pest-parser.github.io) / [site](https://github.com/pest-parser/site)) code appears to depend on `stdweb` which doesn't seem to be maintained any more: https://rustsec.org/advisories/RUSTSEC-2020-0056.html So, I guess it may make sense to go with a more comprehensive re-work (e.g. using [wasm-bindgen](https://rustwasm.github.io/wasm-bindgen/)).

**Comment by tomtau** [comment]
The new release is out: https://github.com/pest-parser/pest/discussions/669
A few notes:
- We found a `rust-version` discrepancy during the release process: https://github.com/pest-parser/pest/issues/671
- `pest` and `pest_meta` were released via https://github.com/katyo/publish-crates GH Action. `pest_vm` mysteriously failed with the error `failed to select a version for the requirement pest_meta = "^2.2.0"` despite `pest_meta 2.2.0` being previously uploaded... it seems it may be due to this bug: https://github.com/katyo/publish-crates/issues/291
- I then published the remaining crates (`pest_vm`, `pest_generator`, `pest_derive`, `pest_grammars`) manually.

**Reply by tomtau** [reply]
Hm, and a breaking change: https://github.com/pest-parser/pest/issues/672
